### ISSUE TITLE
How to use Rocket-nginx with LXD Containers and a reverse proxy

### ISSUE THREAD

**Issue by AlexandruSerban142** [maximejobin/rocket-nginx#69]
Hello, I have a google cloud compute engine instance working with multiple lxd containers described as follow:
Container #1 - Nginx Reverse proxy with two vhosts with proxy_pass to Container#2 and #3 
Container #2 - Nginx, Php7.1-fpm, mysql, phpmyadmin, wordpress, wp-rocket
Container #3 - Same as Container#2

The SSL is finished at the Container #1 level.

Question 1 - Would you recommend the configuration above or delete wp-rocket and use proxy_cache at Container #1 for the other 2 container? Why?
Question 2  - Instead of having Rocket-Nginx installed at both Container #2 and #3, would it be possible to install it only at Container #1 ? Any advantages?
Question 3 - Any more recommendations to my setup?

Thx a lot and thank you for the great module.

**Comment by maximejobin** [comment]
Question 1 - I would use WP-Rocket because it does more than just caching the files. It minifies them and you can also flush cache easily or tell it to not put in cache certain pages. Doing that with the proxy cache requires work I wouldn't want to do.

Question 2 - I would personally install Rocket-Nginx at the first level. Sooner the better. That way, you reduce the footprint.

Question 3 - Is "Container 1" your way to have a load balancer that you manager yourself ? Otherwise, why don't you use a load balancer to do that ?  I don't use Google cloud so maybe what I'm saying does not even make sense. I'm used to AWS where you would use ELB and not create the load balancer yourself.

I hope it helps! :-)

**Comment by AlexandruSerban142** [comment]
Question 1 - Thx, I'll stick with that.

Question 2 - I thought as well that would be better to install rocket-nginx at first level but how can I do it? I have Wordpress + Nginx installed just on Container #2 and #3.

Question 3 - Container #1 with nginx isn't in fact used for the moment as a load balancer because I don't need that.  I use it so  Domain1.com --> Container #1 and Domain2 --> Container #2. Can a load balancer do that? I am just starting to learn about servers, is it possible to use a load balancer to map multiple domains to containers in the same server? In my nginx reverse proxy Container # 1 I just created 2 configs with proxy_pas to the internal Ip's of the Container #2 and #3. 

Google Cloud does have load balancers but I thought that those are used when your server isn't having enough resources and the load balancers loads another identical VM as the first one to redirect traffic to the second VM as well. In my case I have 2 Domains and 1 VM. Can I use a load balancer for what I need? Any more tips? 
Thx a lot for the prompt response and for the great help. Have a nice day!

**Comment by maximejobin** [comment]
Let me rephrase question 2's answer. Despite the fact it would be the best option to have Rocket-Nginx at the first level, this is not possible because there is logic and it can't be achieved at 2 different levels.

As for load balancers, they are usually used simply to split the traffic between all your servers.  I don't know your exact configuration so I won't give you advice. That being said, it looks like your Container 1 acts as a load balancer.
